docs/development/getting-started-locally-fakegcs.md

```diff
@@ -0,0 +1,93 @@
+# Setting up etcd-druid locally with FakeGCS and Kind
+
+This guide provides step-by-step instructions on how to set up etcd-druid with [fake-gcs](https://github.com/fsouza/fake-gcs-server) and Kind on your local machine. FakeGCS emulates the GCS services locally, which allows the etcd cluster to interact with GCS without the need for an actual GCP connection. This setup is ideal for local development and testing.
+
+## Prerequisites
+
+- Docker (installed and running)
+- [gsutil](https://cloud.google.com/storage/docs/gsutil) 
+
+## Environment Setup
+
+### Step 1: Provision the Kind Cluster
+
+Execute the command below to provision a `kind` cluster. This command also forwards port `8000`(HTTP) and `4443`(HTTPS) from the [kind cluster](../../hack/e2e-test/infrastructure/kind/cluster.yaml) to your local machine, enabling access to fake-gcs running inside a pod in the KIND cluster:
+
+```bash
+make kind-up
+```
+
+### Step 2: Deploy FakeGCS
+
+Deploy FakeGCS onto the Kubernetes cluster using the command below:
+
+```bash
+make deploy-fakegcs
+```
+
+### Step 3: Create a GCS bucket for etcd-druid backup purposes
+
+```bash
+gsutil -o "Credentials:gs_json_host=127.0.0.1" -o "Credentials:gs_json_port=4443" -o "Boto:https_validate_certificates=False" mb "gs://etcd-bucket"
+```
+
+### Step 4: Deploy etcd-druid
+
+> **Note :** To avoid file storage permission issues when using `fakegcs`, use [etcd wrapper](https://github.com/gardener/etcd-wrapper). Refer the [ops-guide](https://github.tools.sap/kubernetes/ops-guide/blob/master/01_Operations/02_How_To_Guides/03_Other/ETCD_Operations.md#debugging-etcd-clusters-running-a-etcd-wrapper-images) to learn how to debug etcd-wrapper
+
+To use etcd-wrapper for druid local setup, export the following flag:
+
+```bash
+export USE_ETCD_DRUID_FEATURE_GATES=true 
+```
+
+Deploy etcd-druid onto the Kind cluster using the command below:
+
+```bash
+make deploy
+```
+
+### Step 5: Configure etcd with FakeGCS object store
+
+To configure `FakeGCS` to be used as backup store for etcd, Uncomment the `spec.backup.store` section of the etcd yaml in `config/samples/druid_v1alpha1_etcd.yaml` and replace it with the following:
+
+```yaml
+store:
+  secretRef:
+  name: etcd-backup-gcp
+  container: etcd-bucket
+  provider: gcp # options: aws,azure,gcp,openstack,alicloud,dell,openshift,local
+  prefix: etcd-test
+```
+
+Now, apply the required Kubernetes manifests to create an etcd custom resource (CR) and a secret for GCP credentials, facilitating FakeGCS access:
+
+
+
+```bash
+export KUBECONFIG=hack/e2e-test/infrastructure/kind/kubeconfig
+kubectl apply -f config/samples/druid_v1alpha1_etcd.yaml -f config/samples/etcd-secret-fakegcs.yaml
+```
```

Comment:
I followed the steps provided, but a full snapshot was not taken. Additionally, I observed errors in the backup-restore logs indicating issues with accessing the GCS JSON credential file. Here are the relevant log entries:

```bash
backup-restore time="2024-07-16T09:25:42Z" level=warning msg="Taking scheduled full snapshot failed: error checking if the credentials were updated error checking the modification time of the access credentials failed to fetch file information of the GCS JSON credential file /var/.gcp/serviceaccount.json with error: stat /var/.gcp/serviceaccount.json: no such file or directory" actor=snapshotter
backup-restore time="2024-07-16T09:25:42Z" level=error msg="Failed to take substitute first full snapshot: error checking if the credentials were updated error checking the modification time of the access credentials failed to fetch file information of the GCS JSON credential file /var/.gcp/serviceaccount.json with error: stat /var/.gcp/serviceaccount.json: no such file or directory" actor=backup-restore-server
backup-restore time="2024-07-16T09:25:42Z" level=info msg="Probing etcd..." actor=backup-restore-server
backup-restore time="2024-07-16T09:25:42Z" level=info msg="Probing Etcd by checking etcd status ..." actor=backup-restore-server
backup-restore time="2024-07-16T09:25:42Z" level=info msg="Setting status to : 200" actor=backup-restore-server
backup-restore time="2024-07-16T09:25:42Z" level=info msg="Taking scheduled full snapshot for time: 2024-07-16 09:25:42.462752334 +0000 UTC" actor=snapshotter
```

It appears that the GCS JSON credential file is missing or incorrectly configured.

Reply:
Earlier in the `etcd-secret-fakegcs.yaml` file, the `data.serviceaccount.json` field wrongly had hashtag in front of the encoded value like `#ZHVtbXk= #dummy`. Now I've removed that it should work fine.